test/integration/taskBasedStatusUpdate.test.js

```diff
@@ -287,67 +287,117 @@ describe("Task Based Status Updates", function () {
   });
 
   describe("PATCH Integration tests for Changing the status to IDLE based on users list passed", function () {
-    let superUserId;
-    let superUserJwt;
+    let userId0;
     let userId1;
     let userId2;
     let userId3;
     let userId4;
     let userId5;
+    let userId6;
+    let userId7;
+    let userId8;
+    let userId9;
+    let superUserJwt;
     let listUsers;
     const reqBody = {};
 
     beforeEach(async function () {
-      superUserId = await addUser(userData[4]);
-      superUserJwt = authService.generateAuthToken({ userId: superUserId });
-
-      userId1 = await addUser(userData[6]);
-      userId2 = await addUser(userData[8]);
-      userId3 = await addUser(userData[9]);
-      userId4 = await addUser(userData[0]);
-      userId5 = await addUser(userData[1]);
-      listUsers = [userId1, userId2, userId3, userId4, userId5];
+      userId0 = await addUser(userData[0]);
+      userId1 = await addUser(userData[1]);
+      userId2 = await addUser(userData[2]);
+      userId3 = await addUser(userData[3]);
+      userId4 = await addUser(userData[4]);
+      userId5 = await addUser(userData[5]);
+      userId6 = await addUser(userData[6]);
+      userId7 = await addUser(userData[7]);
+      userId8 = await addUser(userData[8]);
+      userId9 = await addUser(userData[9]);
+      superUserJwt = authService.generateAuthToken({ userId: userId4 });
+      listUsers = [
+        { userId: userId0, expectedState: "IDLE" },
+        { userId: userId1, expectedState: "IDLE" },
+        { userId: userId2, expectedState: "IDLE" },
+        { userId: userId3, expectedState: "IDLE" },
+        { userId: userId4, expectedState: "IDLE" },
+        { userId: userId5, expectedState: "ACTIVE" },
+        { userId: userId6, expectedState: "ACTIVE" },
+        { userId: userId7, expectedState: "ACTIVE" },
+        { userId: userId8, expectedState: "ACTIVE" },
+        { userId: userId9, expectedState: "ACTIVE" },
+      ];
       reqBody.users = listUsers;
-      await userStatusModel.doc("userStatus001").set(generateStatusDataForState(userId1, userState.ACTIVE));
-      await userStatusModel.doc("userStatus002").set(generateStatusDataForState(userId2, userState.OOO));
-      await userStatusModel.doc("userStatus003").set(generateStatusDataForState(userId3, userState.IDLE));
-      await userStatusModel.doc("userStatus004").set(generateStatusDataForState(userId4, userState.ONBOARDING));
+      await userStatusModel.doc("userStatus000").set(generateStatusDataForState(userId0, userState.ACTIVE));
+      await userStatusModel.doc("userStatus001").set(generateStatusDataForState(userId1, userState.OOO));
+      await userStatusModel.doc("userStatus002").set(generateStatusDataForState(userId2, userState.IDLE));
+      await userStatusModel.doc("userStatus003").set(generateStatusDataForState(userId3, userState.ONBOARDING));
+      await userStatusModel.doc("userStatus005").set(generateStatusDataForState(userId5, userState.ACTIVE));
```

Comment:
Nit: userStatus003 then userStatus004 :)

Reply:
I had tried to map userIdx with userStatus00x hence. for userId4 the user status doc didnt exist hence It will be created by system hence skipping it here